diffrax/integrate.py

```diff
@@ -830,23 +1025,42 @@ def _promote(yi):
     ts = jnp.full(out_size, jnp.inf)
     ys = jtu.tree_map(lambda y: jnp.full((out_size,) + jnp.shape(y), jnp.inf), y0)
     result = jnp.array(RESULTS.successful)
+
     if saveat.dense:
         error_if(t0 == t1, "Cannot save dense output if t0 == t1")
         if max_steps is None:
             raise ValueError(
                 "`max_steps=None` is incompatible with `saveat.dense=True`"
             )
-        (
-            _,
-            _,
-            dense_info,
-            _,
-            _,
-        ) = solver.step(terms, tprev, tnext, y0, args, solver_state, made_jump)
+        else:
+            if delays is not None:
+                (
+                    _,
+                    _,
+                    dense_info,
+                    _,
+                    _,
+                ) = solver.step(terms_, tprev, tnext, y0, args, solver_state, made_jump)
+
+                new_discontinuity = jnp.full(200 + 1, jnp.inf)
+                new_discontinuity = new_discontinuity.at[
```

Comment:
Instead of instantiating a `max_steps` array i gave `200`. This speeds up integration process

Reply:
Let's introduce a  `max_discontinuities` argument here for the size of this.

If `discontinuities` is passed then let's just use that directly, as well. (i.e. skip the `.at[].set()`.)

Reply:
For https://github.com/patrick-kidger/diffrax/pull/169#discussion_r991711389 , `discontinuity` is more of an argument of the DDE, it would give the initial discontinuities of the history function and the initial point. Maybe to avoid confusion, I could go for `initial_discontinuity` ? 

 Ok will pass a ` max_discontinuities` arg